packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -182,8 +181,19 @@ type StoreConsistencyCheck<T> = {
   getSnapshot: () => T,
 };
 
+type EventFunctionState<T> = {
+  event: EventFunctionWrapper<T>,
+  nextEvent: EventFunction<T>,
+};
+type EventFunctionWrapper<T> = {
+  (): T,
+  _current: EventFunction<T>,
```

Comment:
maybe this could be called `_impl` and then `nextEvent` could be called `nextImpl`